### PR TITLE
feat(drawer) Adding default Web MDC temporary drawer close behavior

### PR DESCRIPTION
**Pull request by PlatanoBailando**
Comparing the behavior of the MDC Web Temporary Drawer with Angular MDC Web's Temporary Drawer, there is an inconsistency.  In particular, the MDC Web version does not close upon clicking inside the drawer, while Angular MDC Web's does.  This is achieved in MDC Web via a stopPropogation call inside the drawer.

This PR, instead of converting directly to the behavior in MDC Web, adds a property to the temporary drawer which toggles between the two behaviors, with the default being what Angular MDC Web currently does (i.e. close on a click anywhere).

### CODE REVIEW

**Comment by trimox** [comment]
Hey @PlatanoBailando 
Thanks for your PR! I really like this idea of supporting both behavior patterns for temporary drawer navigation 👍 

Will review and let ya know

**Comment by gline9** [comment]
Good work on the drawer Chris!

**Comment by PlatanoBailando** [comment]
@trimox I realized that the original commit didn't fix the underlying issue, nor was it in line with the rest of the code.  I've updated it to reflect this.

I believe the underlying issue is that MDC Web's (de)registerDrawerInteractionHandler methods expect to be manipulating events on the drawer dom element itself.  (See https://github.com/material-components/material-components-web/blob/master/packages/mdc-drawer/temporary/index.js)  This differs from Angular MDC Web's in that it's attaching everything to 'root', which is why the drawer wasn't closing in the first place.  The advantage of the new commit is getting new event handlers from MDC Web if they add such to the drawer itself, in addition to being more in line with the rest of the code.

**Comment by trimox** [comment]
@PlatanoBailando Thanks for the update. Preparing a quick v0.5.5 release today, afterwhich I'll review and test PR.

File: src/demo-app/components/drawer-demo/drawer-demo.html [hunk 1]
```diff
@@ -167,6 +167,18 @@ <h1 mdc-typography-display1>Drawers</h1>
       </tbody>
       <thead>
         <tr>
+          <th>Properties</th>
+          <th>Description</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td>closeOnClick</td>
```
Comment (trimox) [inline src/demo-app/components/drawer-demo/drawer-demo.html]: Minor nit: Need a property type : `closeOnClick: boolean`

File: src/lib/drawer/temporary/drawer-temporary.component.ts [hunk 2]
```diff
@@ -89,6 +88,10 @@ export class MdcTemporaryDrawerComponent implements AfterViewInit, OnDestroy {
   @Output() closed: EventEmitter<void> = new EventEmitter<void>();
   @HostBinding('class.mdc-temporary-drawer') isHostClass = true;
   @ViewChild(MdcTemporaryDrawerNavigationDirective) drawerNav: MdcTemporaryDrawerNavigationDirective;
+  @Input() closeOnClick: boolean = true;
+  private get drawerElement() {
```
Comment (trimox) [inline src/lib/drawer/temporary/drawer-temporary.component.ts]: Nice 👍

**Comment by PlatanoBailando** [comment]
@trimox  Glad you like it.  I've added the property type to the documentation.

**Comment by trimox** [comment]
LGTM! Thank you!!

**Pull request merged**